src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -271,15 +273,29 @@ IFACEMETHODIMP_(void) FancyZones::WindowCreated(HWND window) noexcept
 {
     if (m_settings->GetSettings().appLastZone_moveWindows)
     {
-        auto processPath = get_process_path(window);
-        if (!processPath.empty())
+        auto monitor = MonitorFromWindow(window, MONITOR_DEFAULTTONULL);
+        if (monitor)
         {
-            const auto& fancyZonesData = JSONHelpers::FancyZonesDataInstance();
-            int zoneIndex = fancyZonesData.GetAppLastZone(window, processPath.data());
-
-            if (zoneIndex != -1)
+            auto zoneWindow = m_zoneWindowMap.find(monitor);
+            if (zoneWindow != m_zoneWindowMap.end())
             {
-                MoveWindowIntoZoneByIndex(window, zoneIndex);
+                const auto& zoneWindowPtr = zoneWindow->second;
+                const auto activeZoneSet = zoneWindowPtr->ActiveZoneSet();
+                if (activeZoneSet)
+                {
+                    const auto& fancyZonesData = JSONHelpers::FancyZonesDataInstance();
+
+                    OLECHAR* guidString;
+                    StringFromCLSID(activeZoneSet->Id(), &guidString);

```

Comment:
For APIs that return HRESULT, it's better to check it instead of assuming they will always succeed.